azkaban-hadoop-security-plugin/src/main/java/azkaban/security/ExecuteAsUser.java

```diff
@@ -0,0 +1,116 @@
+package azkaban.security;
+
+import org.apache.log4j.Logger;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+
+/**
+ *
+ * Created by spyne on 9/20/16.
+ */
+public class ExecuteAsUser {
+  private final static Logger log = Logger.getLogger(ExecuteAsUser.class);
+
+  private final String path;
+
+  public ExecuteAsUser(final String path) {
+    validate(path);
+    this.path = path;
+  }
+
+  private void validate(String path) {
+    File executeAsUserBinary = new File(path);
+    if (!executeAsUserBinary.canExecute()) {
+      throw new RuntimeException("Unable to execute execute-as-user binary. Invalid Path: " + path);
+    }
+  }
+
+  public int execute(final String user, final String command) throws IOException {
+    log.info("Command: " + command);
+    Process process = new ProcessBuilder()
+        .command(partitionCommandLine(constructExecuteAsCommand(user, command)))
+        .inheritIO()
+        .start();
+
+    int exitCode;
+    try {
+      exitCode = process.waitFor();
+    } catch (InterruptedException e) {
+      e.printStackTrace();
+      exitCode = 1;
+    }
+    return exitCode;
+  }
+
+  private String constructExecuteAsCommand(String user, String command) {
+    return String.format("%s %s %s", path, user, command);
+  }
+
+  /**
+   * TODO Refactor: function copied from azkaban.jobExecutor.ProcessJob#partitionCommandLine(java.lang.String)
+   *
+   * Need to be refactored to use just one method. Copy was created to avoid introducing dependencies to
+   * azkaban-common
+   *
+   * Splits the command into a unix like command line structure. Quotes and
+   * single quotes are treated as nested strings.
```

Comment:
Updated it. I didn't modify any piece of that code. Hence I didn't bother to add tests. Moreover, I think this piece of code should be handled by a library as opposed to doing this by hand. I don't find it to be maintainable.